src/core/Object3D.js

```diff
@@ -589,13 +589,17 @@ class Object3D extends EventDispatcher {
 
 		if ( this.matrixWorldNeedsUpdate || force ) {
 
-			if ( this.parent === null ) {
+			if ( this.matrixWorldAutoUpdate === true ) {
```

Comment:
How about adding the same check of `matrixWorldAutoUpdate` to `updateWorldMatrix()` as well?

Ideally, both methods should behave the same meaning when `matrixWorldAutoUpdate` is set to `false`, no world matrix computation happens.

Besides,  `updateWorldMatrix()` has the same bug like `updateMatrixWorld()` that should be fixed with the PR as well. When calling `updateWorldMatrix()` on the parent or child reference, `matrixWorldAutoUpdate` should _not_ be checked. Otherwise it prevents the update of ancestor or descendants that do set `matrixWorldAutoUpdate` to `true`.

Ultimately, the method should look like so:

```
updateWorldMatrix( updateParents, updateChildren ) {

	const parent = this.parent;

	if ( updateParents === true && parent !== null ) {

		parent.updateWorldMatrix( true, false );

	}

	if ( this.matrixAutoUpdate ) this.updateMatrix();

	if ( this.matrixWorldAutoUpdate === true ) {

		if ( this.parent === null ) {

			this.matrixWorld.copy( this.matrix );

		} else {

			this.matrixWorld.multiplyMatrices( this.parent.matrixWorld, this.matrix );

		}

	}

	// update children

	if ( updateChildren === true ) {

		const children = this.children;

		for ( let i = 0, l = children.length; i < l; i ++ ) {

			const child = children[ i ];

			child.updateWorldMatrix( false, true );

		}

	}

}
```

Reply:
Let's add a test for this. Interesting we don't check for `matrixWorldNeedsUpdate` here so that `updateParents` will work.

I wonder if starting from the root down will better serve #28534. For a future PR if any.